step/step.go

```diff
@@ -0,0 +1,121 @@
+package step
+
+import (
+	"fmt"
+	"os"
+
+	"github.com/bitrise-steplib/bitrise-step-android-instrumented-test/apk_info"
+
+	"github.com/bitrise-io/go-android/v2/adbmanager"
+	"github.com/bitrise-io/go-android/v2/sdk"
+	"github.com/bitrise-io/go-steputils/v2/stepconf"
+	"github.com/bitrise-io/go-utils/v2/command"
+	"github.com/bitrise-io/go-utils/v2/log"
+	"github.com/kballard/go-shellquote"
+)
+
+type Input struct {
+	MainAPKPath              string `env:"main_apk_path,required"`
+	TestAPKPath              string `env:"test_apk_path,required"`
+	TestRunnerClass          string `env:"test_runner_class,required"`
+	AdditionalTestingOptions string `env:"additional_testing_options"`
+}
+
+type Config struct {
+	MainAPKPath              string
+	TestAPKPath              string
+	TestRunnerClass          string
+	AdditionalTestingOptions []string
+}
+
+type InstrumentedTestRunner struct {
+	logger         log.Logger
+	inputParser    stepconf.InputParser
+	commandFactory command.Factory
+}
+
+func New(
+	logger log.Logger,
+	inputParser stepconf.InputParser,
+	commandFactory command.Factory,
+) InstrumentedTestRunner {
+	return InstrumentedTestRunner{
+		logger:         logger,
+		inputParser:    inputParser,
+		commandFactory: commandFactory,
+	}
+}
+
+func (testRunner InstrumentedTestRunner) ProcessConfig() (*Config, error) {
+	var input Input
+	if err := testRunner.inputParser.Parse(&input); err != nil {
+		return nil, fmt.Errorf("unable to parse input: %w", err)
```

Comment:
no need to warp the error of the input parser, since the original error looks like this:

```
failed to parse config:
- <issue_1>
- <issue_2>
...
```

The current `unable to parse input` does not add too much useful detail to the error:

```
unable to parse input: failed to parse config:
- <issue_1>
- <issue_2>
...
```